File: src/Entity/Choix.php
<?php

namespace App\Entity;

use App\Repository\ChoixRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


/**
 * @ORM\Entity(repositoryClass=ChoixRepository::class)
 */
class Choix
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="boolean")
     */
    private $etatchoix;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Aucun choix peut être null")
     */
    private $contenu;

    /**
     * @ORM\ManyToOne(targetEntity=Question::class, inversedBy="choix")
     */
    private $question;

    /**
     * @ORM\ManyToOne(targetEntity=Quiz::class, inversedBy="choix")
     */
    private $quiz;

    /**
     * @ORM\Column(type="boolean", nullable=true)
     */
    private $checked;

    

    

   

    


    
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getEtatChoix(): ?bool
    {
        return $this->etatchoix;
    }

    public function setEtatChoix(bool $etatchoix): self
    {
        $this->etatchoix = $etatchoix;

        return $this;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): self
    {
        $this->contenu = $contenu;

        return $this;
    }
    

    public function getQuestion(): ?Question
    {
        return $this->question;
    }

    public function setQuestion(?Question $question): self
    {
        $this->question = $question;

        return $this;
    }

    public function getQuiz(): ?Quiz
    {
        return $this->quiz;
    }

    public function setQuiz(?Quiz $quiz): self
    {
        $this->quiz = $quiz;

        return $this;
    }

    public function getChecked(): ?bool
    {
        return $this->checked;
    }

    public function setChecked(?bool $checked): self
    {
        $this->checked = $checked;

        return $this;
    }

    


    


    

    
}

File: CRUDWorkshop(si) (1)/CRUDWorkshop(si) (1)/CRUDWorkshop(si)/src/edu/esprit/entities/Choix.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.esprit.entities;

/**
 *
 * @author Moez
 */
public class Choix {
    
    private int id;
    private String contenu;
    private int etatchoix;
    private int checked;
    private Question question_ch;

    public Choix() {
    }

    public Choix(int id, String contenu, int etatchoix) {
        this.id = id;
        this.contenu = contenu;
        this.etatchoix = etatchoix;
    }

    public Choix(int id, String contenu, int etatchoix, int checked, Question question_ch) {
        this.id = id;
        this.contenu = contenu;
        this.etatchoix = etatchoix;
        this.checked = checked;
        this.question_ch = question_ch;
    }
    
    public Choix(String contenu, int etatchoix, int checked, Question question_ch) {
        this.id = id;
        this.contenu = contenu;
        this.etatchoix = etatchoix;
        this.checked = checked;
        this.question_ch = question_ch;
    }
    
    
    public Choix(int id,int etatchoix, int checked) {
        this.id = id;
        this.etatchoix = etatchoix;
        this.checked = checked;
    }
    

    public Choix(String contenu,Question question_ch) {
        this.contenu = contenu;
        this.etatchoix = etatchoix;
        this.question_ch = question_ch;
    }
    
    
    
    public Choix(String contenu, int etatchoix) {
        this.contenu = contenu;
        this.etatchoix = etatchoix;
    }
    
    public Choix(int id,String contenu) {
        this.id = id;
        this.contenu = contenu;
    }

    public String getContenu() {
        return contenu;
    }

    public int getEtatchoix() {
        return etatchoix;
    }

    public int getId() {
        return id;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public void setEtatchoix(int etatchoix) {
        this.etatchoix = etatchoix;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getChecked() {
        return checked;
    }

    public void setChecked(int checked) {
        this.checked = checked;
    }
    
    
    
    
    
    
    
}
